### PR TITLE
Update HK version to fix issue with db-init-script

### LINKED ISSUE #111: Housekeeping requires `circus_train` database
I discovered an issue with changing the default Housekeeping schema name from `circus_train` to anything else.

Test command:
```/opt/circus-train/bin/circus-train.sh --config=/mnt/circus-train/jobs/test.yml --modules=replication --housekeeping.data-source.username=$HK_USER --housekeeping.data-source.password=$HK_PASSWORD```

Test YAML file:
```---
  instance:
    name: dsp-lab-jetstream-ct
    home: /mnt/circus-train
  logging:
    config: file:${instance.home}/conf/log4j.xml
  housekeeping:
    schema-name: housekeeping
    data-source:
      driver-class-name: com.mysql.cj.jdbc.Driver
      url: jdbc:mysql://jetstream-db.dsp-lab.aws:3306/housekeeping
  copier-options: 
    tmp-dir: "hdfs:///tmp/dsp-lab-jetstream-ct/"
    file-attribute: "replication, blocksize, user, group, permission, checksumtype"
    canned-acl: "bucket-owner-full-control"
    region: "us-west-2"
    max-maps: 50
    s3-server-side-encryption: true
    copier-factory-class: "com.hotels.bdp.circustrain.s3mapreducecpcopier.S3MapReduceCpCopierFactory"
  source-catalog: 
    name: "mauihdp"
    hive-metastore-uris: "thrift://cheledwhdm006.karmalab.net:9083"
  replica-catalog: 
    name: "apiary-lab-hms-us-west-2"
    hive-metastore-uris: "thrift://hms-readwrite.apiary-us-west-2.lcl:9083"
  table-replications: 
    - 
      source-table: 
        database-name: "dm"
        table-name: "ar_typ_dim"
        generate-partition-filter: true
      replica-table: 
        database-name: "dm"
        table-name: "ar_typ_dim"
        table-location: "s3://apiary-470557454065-us-west-2-dm/ar_typ_dim"
  sns-event-listener: 
    region: "us-west-2"
    topic: "arn:aws:sns:us-west-2:470557454065:jetstream-lambda-sns"
    subject: "CircusTrainStatus"
    headers: 
      requestId: "api13cda12a-18f8-11e9-9be1-abf8ece1bc04"
      databaseTableName: "dm.ar_typ_dim"
      route: "EXPEDIA-DSP-MAUI-ETL.EXPEDIA-DSP-LAB-UW2-APIARY"```

As shown above, housekeeping has been reconfigured to use a schema named `housekeeping` in a MySQL server.

When this schema does not exist, Circus Train fails with the following error:

```19/01/15 11:16:13 ERROR boot.SpringApplication: Application startup failed
org.springframework.beans.factory.BeanCreationException: Error creating bean with name 'org.springframework.boot.autoconfigure.orm.jpa.HibernateJpaAutoConfiguration': Injection of autowired dependencies failed; nested exception is org.springframework.beans.factory.BeanCreationException: Could not autowire field: private javax.sql.DataSource org.springframework.boot.autoconfigure.orm.jpa.JpaBaseConfiguration.dataSource; nested exception is org.springframework.beans.factory.BeanCreationException: Error creating bean with name 'housekeepingDataSource' defined in class path resource [com/hotels/housekeeping/HousekeepingConfiguration.class]: Initialization of bean failed; nested exception is org.springframework.beans.factory.BeanCreationException: Error creating bean with name 'dataSourceInitializer': Invocation of init method failed; nested exception is org.springframework.jdbc.datasource.init.ScriptStatementFailedException: Failed to execute SQL script statement #1 of class path resource [schema.sql]: CREATE SCHEMA IF NOT EXISTS circus_train; nested exception is com.mysql.jdbc.exceptions.jdbc4.MySQLSyntaxErrorException: Access denied for user 'jetstream_rw'@'%' to database 'circus_train'```
[ct-failed.txt](https://github.com/HotelsDotCom/circus-train/files/2761166/ct-failed.txt)

After creating `circus_train` schema, the above Circus Train command starts to work.  It also uses the configured `housekeeping` schema instead of the new `circus_train` schema:

```mysql> use circus_train;
Database changed

mysql> show tables;
Empty set (0.00 sec)

mysql> use housekeeping;
Database changed

mysql> show tables;
+------------------------+
| Tables_in_housekeeping |
+------------------------+
| audit_revision         |
| legacy_replica_path    |
+------------------------+
2 rows in set (0.01 sec)```

It appears that the initial schema check ignores the configured `housekeeping.schema-name` value, and instead always checks for the `circus_train` schema.  But if that schema does exist, Circus Train skips the `dataSourceInitializer` and the rest of the housekeeping code uses the correct schema as configured.

### PR DESCRIPTION
**Pull request by AnanaMJ**
Fixes #111 

The main change is the README update and the update of housekeeping version.

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -4,7 +4,7 @@
   <parent>
     <groupId>com.hotels</groupId>
     <artifactId>hotels-oss-parent</artifactId>
-    <version>2.3.5</version>
+    <version>3.0.0</version>
```
Comment (AnanaMJ) [inline pom.xml]: If houskeeping is fine for Jenkins with `hotels-oss-parent` 4.0.0, then I'll update this to v4 as well.

File: pom.xml [hunk 2]
```diff
@@ -61,7 +62,7 @@
     <derby.version>10.10.2.0</derby.version>
     <beeju.version>1.2.1</beeju.version>
     <hamcrest-library.version>1.3</hamcrest-library.version>
-    <housekeeping.version>3.0.5</housekeeping.version>
+    <housekeeping.version>3.0.6-SNAPSHOT</housekeeping.version>
```
Comment (AnanaMJ) [inline pom.xml]: Once housekeeping is released, I'll update this to 3.0.6, otherwise the version won't be resolved.

File: README.md [hunk 3]
```diff
@@ -813,4 +815,4 @@ The Circus Train logo uses the [Ewert font](http://www.1001fonts.com/ewert-font.
 # Legal
 This project is available under the [Apache 2.0 License](http://www.apache.org/licenses/LICENSE-2.0.html).
 
-Copyright 2016-2019 Expedia Inc.
+Copyright 2016-2019 Expedia Group
```
Comment (abhimanyugupta07) [inline README.md]: 👍

Comment (abhimanyugupta07) [reply README.md]: Not sure if you want to make that same change in all the files in that case ?

Comment (AnanaMJ) [reply README.md]: Oh no, this is in all the licence headers, right? 😰

Comment (abhimanyugupta07) [reply README.md]: Yeah. I guess, just leave it as is for now to keep it consistent with other BDP projects.

Comment (AnanaMJ) [reply README.md]: But this being in the README is fine?

File: CHANGELOG.md [hunk 4]
```diff
@@ -1,7 +1,12 @@
 ## TBD
+### Changed
+* Updated `housekeeping` version to 3.0.6 (was 3.0.5).
+* Updated `hotels-oss-parent` version to 4.0.0 (was 2.3.5).
```
Comment (massdosage) [inline CHANGELOG.md]: <3

File: README.md (hunk 3 above)
Comment (massdosage) [reply README.md]: I haven't received any guidance on whether this should remain `Inc.` or be changed to `Group` since the name change so I was just leaving everything as `Inc.` for now. I'll ask and let you all know.

File: pom.xml [hunk 5]
```diff
@@ -322,4 +323,17 @@
     </pluginManagement>
   </build>
 
+  <reporting>
```
Comment (massdosage) [inline pom.xml]: Why did you need to add this?

Comment (AnanaMJ) [reply pom.xml]: It was removed in `hotels-oss-parent`

Comment (massdosage) [reply pom.xml]: Ah yes, it doesn't work for Java 8 but since this is Java 7 (for now) we still need it here, OK.

File: README.md (hunk 3 above)
Comment (AnanaMJ) [reply README.md]: Ok, I changed it back to `Expedia Inc.` 🙂
